public/app/ui/pages/search/SearchPage.js

```diff
@@ -25,21 +25,22 @@
                     submittedQuery = ko.observable(),
                     results = ko.observableArray(),
                     query = qs.parse(window.location.search.replace(/^\?/, '')),
+                    overlay = container.resolve('ui.overlay'),
                     doSearch = function (callback) {
                         submittedQuery(this.searchText());
                         results([]);
-                        this.loading(true);
+                        overlay.loading(true);
                         this.displayResults(false);
                         container.resolve('search.' + context.params.type)(
                             _.zipObject(
                                 [ selectedSearchType() ],
                                 [ submittedQuery() ]
                             ),
                             function (err, searchServiceResults) {
-                                this.loading(false);
+                                overlay.loading(false);
```

Comment:
This is an additional call to `overlay.loading()` because we have just finished loading search results.